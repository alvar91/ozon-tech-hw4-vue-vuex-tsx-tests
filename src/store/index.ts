import Vue from "vue";
import Vuex from 'vuex';
import { createVuexStore } from "vuex-simple";

// @ts-ignore
import { MainModule } from "./modules/main/main.ts";

Vue.use(Vuex);

const store = createVuexStore(new MainModule());

export default store;
