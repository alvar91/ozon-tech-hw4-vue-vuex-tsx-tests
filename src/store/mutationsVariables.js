export const FETCH_USERS_ACTION = "FETCH_USERS_ACTION";
export const FETCH_USERS = "FETCH_USERS";
export const SET_LOADING = "SET_LOADING";
export const SET_CURRENT_USER = "SET_CURRENT_USER";
export const SET_FILTER_KEYWORD = "SET_FILTER_KEYWORD";
export const SET_FORM_INPUT = "SET_FORM_INPUT";
export const SET_FORM_CHANGE = "SET_FORM_CHANGE";
export const SET_FORM_RADIO = "SET_FORM_RADIO";
