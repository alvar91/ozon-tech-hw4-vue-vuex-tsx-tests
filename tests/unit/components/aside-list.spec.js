import Vuex from "vuex";
import { mount, createLocalVue } from "@vue/test-utils";

import AsideList from "@/components/aside-list.vue";

import { mockUsersArr } from "../../mock/mockUser";

import { FETCH_USERS_ACTION } from "@/store/mutationsVariables";

import store from "@/store/index";

const localVue = createLocalVue();
localVue.use(Vuex);

describe("AsideList.vue", () => {
  let wrapper;

  beforeEach(async () => {
    await store.dispatch(FETCH_USERS_ACTION);

    wrapper = mount(AsideList, {
      localVue,
      store,
    });
  });

  it("should correctly renders markup", () => {
    expect(wrapper.html()).toMatchSnapshot();
  });

  it("should be called on the second user card and change the store", () => {
    const { id: secondUserMockId } = mockUsersArr[1];
    const secondCardLink = wrapper.findAll(".aside__user-link").at(1);
    secondCardLink.trigger("click");
    expect(secondUserMockId).toBe(store.state.currentUserId);
  });

  afterEach(() => {
    wrapper.destroy();
  });
});
