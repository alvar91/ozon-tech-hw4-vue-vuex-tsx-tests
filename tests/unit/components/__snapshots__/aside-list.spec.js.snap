// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AsideList.vue should correctly renders markup 1`] = `
<ul class="aside__list">
  <li data-id="7add8d25-727b-4db4-bc0f-decd057c0c25" class="aside__item"><a href="#" class="aside__user-link">Агафонов Михаил Дмитриевич</a>
    <div class="aside__status"><button class="aside__user-status aside__user-status--active">Загружена кандидатом</button><time datetime="2020-07-21T10:11:08-03:00" class="aside__user-date">21.07.2020</time></div>
  </li>
  <li data-id="e96baf2b-a4c6-4a18-a376-b7589e2b053e" class="aside__item"><a href="#" class="aside__user-link">Гончаров Владислав Александрович</a>
    <div class="aside__status"><button class="aside__user-status aside__user-status--inactive">Ожидают загрузки</button><time datetime="2022-02-28T10:55:31-03:00" class="aside__user-date">28.02.2022</time></div>
  </li>
  <li data-id="b288f6b6-1bd9-4a1b-9a34-012e29b3a273" class="aside__item"><a href="#" class="aside__user-link">Орлов Николай Константинович</a>
    <div class="aside__status"><button class="aside__user-status aside__user-status--inactive">Ожидают загрузки</button><time datetime="2021-09-15T05:34:50-03:00" class="aside__user-date">15.09.2021</time></div>
  </li>
  <li data-id="99ec59ba-cd5f-4392-b3eb-248f97346b9c" class="aside__item"><a href="#" class="aside__user-link">Виноградов Михаил Владиславович</a>
    <div class="aside__status"><button class="aside__user-status aside__user-status--active">Загружена кандидатом</button><time datetime="2019-10-29T06:20:07-03:00" class="aside__user-date">29.10.2019</time></div>
  </li>
  <li data-id="659d9e5c-5bd4-45d3-9da8-a7ac5a2586ba" class="aside__item"><a href="#" class="aside__user-link">Жукова Анастасия Александровна</a>
    <div class="aside__status"><button class="aside__user-status aside__user-status--active">Загружена кандидатом</button><time datetime="2020-06-21T11:57:46-03:00" class="aside__user-date">21.06.2020</time></div>
  </li>
  <li data-id="3d3e4338-5ecd-4a22-9e8a-7b698b3cfa8c" class="aside__item"><a href="#" class="aside__user-link">Жукова Екатерина Константиновна</a>
    <div class="aside__status"><button class="aside__user-status aside__user-status--active">Загружена кандидатом</button><time datetime="2020-06-19T12:29:52-03:00" class="aside__user-date">19.06.2020</time></div>
  </li>
  <li data-id="422aeaaa-d5df-48a6-8d67-30dd9ce09196" class="aside__item"><a href="#" class="aside__user-link">Агафонов Василий Петрович</a>
    <div class="aside__status"><button class="aside__user-status aside__user-status--active">Загружена кандидатом</button><time datetime="2019-09-02T05:02:29-03:00" class="aside__user-date">02.09.2019</time></div>
  </li>
  <li data-id="b8898728-b999-40f5-a713-fcdb595b7da0" class="aside__item"><a href="#" class="aside__user-link">Гончаров Александр Константинович</a>
    <div class="aside__status"><button class="aside__user-status aside__user-status--active">Загружена кандидатом</button><time datetime="2022-10-20T05:31:20-03:00" class="aside__user-date">20.10.2022</time></div>
  </li>
  <li data-id="019bb508-d222-4524-b93d-5365b10202a8" class="aside__item"><a href="#" class="aside__user-link">Воробьева Евангелина Геннадьевна</a>
    <div class="aside__status"><button class="aside__user-status aside__user-status--inactive">Ожидают загрузки</button><time datetime="2019-07-25T09:17:08-03:00" class="aside__user-date">25.07.2019</time></div>
  </li>
  <li data-id="0f2609f8-5aed-46b6-ad14-7d4079842824" class="aside__item"><a href="#" class="aside__user-link">Агафонов Михаил Александрович</a>
    <div class="aside__status"><button class="aside__user-status aside__user-status--inactive">Ожидают загрузки</button><time datetime="2019-01-22T09:47:04-03:00" class="aside__user-date">22.01.2019</time></div>
  </li>
</ul>
`;
